### PR TITLE
[search] Disable suggests when full name token is entered.

### PR DESCRIPTION
**Pull request by ygorshenin**
@vng PTAL

### CODE REVIEW

File: search/search_query.cpp [hunk 1]
```diff
@@ -874,39 +874,41 @@ void Query::RemoveStringPrefix(string const & str, string & res) const
...
+    else if (token.size() >= m_prefix.size() && StartsWith(token, m_prefix))
+    {
+      prefixMatched = true;
+      fullPrefixMatched = token.size() == m_prefix.size();
+    }
   }
 
   // Name doesn't match prefix - do nothing.
-  if (!prefixMatched)
+  if (!prefixMatched || fullPrefixMatched)
```
Comment (mpimenov) [inline search/search_query.cpp]: Комментарий теперь неправильный.

File: search/search_query.cpp [hunk 2]
```diff
@@ -874,39 +874,41 @@ void Query::RemoveStringPrefix(string const & str, string & res) const
...
-    {
-      suggest += strings::ToUtf8(vName[i]);
-      suggest += " ";
-    }
+  // Appends unmatched result's tokens to the suggestion.
+  for (size_t i = 0; i < tokens.size(); ++i)
+  {
+    if (tokensMatched[i])
+      continue;
+    suggest += strings::ToUtf8(tokens[i]);
```
Comment (mpimenov) [inline search/search_query.cpp]: append

File: search/v2/geocoder.cpp [hunk 3]
```diff
@@ -889,6 +889,11 @@ void Geocoder::MatchRegions(RegionType type)
   }
 }
 
+void Geocoder::EmitResult(MwmSet::MwmId const & mwmId, uint32_t featureId)
```
Comment (mpimenov) [inline search/v2/geocoder.cpp]: Please move it to match the order in the .hpp file.

**Comment by vng** [comment]
LGTM

File: search/search_query.cpp [hunk 4]
```diff
@@ -874,39 +874,41 @@ void Query::RemoveStringPrefix(string const & str, string & res) const
...
+    auto const & token = tokens[i];
+    if (find(m_tokens.begin(), m_tokens.end(), token) != m_tokens.end())
       tokensMatched[i] = true;
-    else
-      if (vName[i].size() >= m_prefix.size() &&
-          StartsWith(vName[i].begin(), vName[i].end(), m_prefix.begin(), m_prefix.end()))
-      {
-        prefixMatched = true;
-      }
+    else if (token.size() >= m_prefix.size() && StartsWith(token, m_prefix))
```
Comment (mpimenov) [inline search/search_query.cpp]: Первое условие лишнее?

File: search/search_query.cpp (hunk 1 above)
Comment (ygorshenin) [inline search/search_query.cpp]: Done

File: search/search_query.cpp (hunk 2 above)
Comment (ygorshenin) [inline search/search_query.cpp]: Done

File: search/v2/geocoder.cpp (hunk 3 above)
Comment (ygorshenin) [inline search/v2/geocoder.cpp]: Done

File: search/search_query.cpp (hunk 4 above)
Comment (ygorshenin) [inline search/search_query.cpp]: Done

**Comment by ygorshenin** [comment]
PTAL @mpimenov

**Comment by mpimenov** [comment]
LGTM

**Pull request merged**